crates/core/tests/full_path.rs

```diff
@@ -0,0 +1,850 @@
+// Verifies that tables with fully-qualified file paths behave as expected
+// and that reading them yields the same data as the corresponding table
+// with relative file paths.
+use serde_json::Value;
+use std::collections::HashMap;
+use std::fs;
+use std::path::{Path, PathBuf};
+use url::Url;
+
+// Small utility to build original/cloned URIs from a base URI and two prefixes
+fn build_prefixed_uris(
+    base_uri: &str,
+    original_prefix: &str,
+    cloned_prefix: &str,
+) -> (String, String) {
+    let base = base_uri.trim_end_matches('/');
+    let original = format!("{}/{}", base, original_prefix);
+    let cloned = format!("{}/{}", base, cloned_prefix);
+    (original, cloned)
+}
+
+// A generic RAII cleanup helper that runs a command on drop. Errors are ignored.
+struct CommandCleanup {
+    program: &'static str,
+    args: Vec<String>,
+}
+
+impl CommandCleanup {
+    fn new(program: &'static str, args: &[&str]) -> Self {
+        Self {
+            program,
+            args: args.iter().map(|s| s.to_string()).collect(),
+        }
+    }
+}
+
+impl Drop for CommandCleanup {
+    fn drop(&mut self) {
+        use std::process::Command;
+        let _ = Command::new(self.program).args(&self.args).status();
+    }
+}
+
+// Test to use absolute file paths
+#[tokio::test]
+async fn compare_table_with_full_paths() {
+    // std::env::set_var("DELTA_RS_ALLOW_UNRESTRICTED_FILE_ACCESS", "1");
```

Comment:
When I run these tests locally this test fails, is that to be expected with this pull request? I.e. is this supposed to be a demonstration of a bug?